saleor/checkout/test_checkout.py

```diff
@@ -164,3 +170,755 @@ def test_index_view(cart, status_code, url, rf):
     response = views.index_view(request, checkout)
     assert response.status_code == status_code
     assert response.url == url
+
+
+empty_cart = Mock(__len__=Mock(return_value=0))
+cart_with_shipping = Mock(__len__=Mock(return_value=1),
+                          is_shipping_required=Mock(return_value=True))
+cart_without_shipping = Mock(__len__=Mock(return_value=1),
+                             is_shipping_required=Mock(return_value=False))
+
+
+@pytest.fixture
+def shipping_method(db):
+    shipping_method = ShippingMethod.objects.create(name='Shipping method')
+    shipping_method_country = ShippingMethodCountry.objects.create(
+        country_code='PL', shipping_method=shipping_method,
+        price=Price('3.0', currency=settings.DEFAULT_CURRENCY))
+    return shipping_method_country
+
+
+@pytest.fixture
+def client_with_cart(client, product_in_stock):
+    product_variant = ProductVariant.objects.get()
```

Comment:
You are explicitly passed `product_in_stock`. There is no need to expect that there will always be exactly one product variant.